spec/forms/school_search_form_spec.rb

```diff
@@ -22,5 +22,30 @@
       schools = form.find_schools(1)
       expect(schools.count).to eq(3)
     end
+
+    it "finds school with matching Unique Reference Number (URN)" do
+      form = SchoolSearchForm.new(school_name: "2345678")
+      schools = form.find_schools(1)
+      expect(schools.count).to eq(1)
+      expect(schools.first.name).to eql("Amazing school")
+    end
+
+    it "finds all schools with an empty query" do
+      form = SchoolSearchForm.new(school_name: "")
+      schools = form.find_schools(1)
+      expect(schools.count).to eq(3)
+    end
+
+    it "finds schools that have a partnership" do
+      school = schools[2]
+      lead_provider = FactoryBot.create(:lead_provider)
+      partnership = FactoryBot.create(:partnership, school: school, lead_provider: lead_provider)
```

Comment:
The purpose of factories is to save us time when setting up for tests. The partnership factory is probably doing the opposite - I would just create a `Partnership` directly. 

And then you can painlessly remove the partnership factory like Alfred suggests :)

Reply:
Makes sense, I wasn't 100% on whether to include the factory bot for a partnership or not so went somewhere in the middle.